### DISCUSSION TITLE
Removing MK1 Monochrome Display

### DISCUSSION THREAD

**Discussion by bow0109** [jakka351/FG-Falcon#6] [Q&A] [answered]
Hi,
I am planning on creating a Raspberry Pi based head unit, similar to what you've done [here](https://github.com/jakka351/can0swc).
Reading up on the ford forums, it seems that once you remove the icc display, you need to send the same canbus codes that the screen did to avoid it throwing a fault code. I found MitchellH's original Arduino code to do this, and I could convert this into python, but was just wondering if you had already done so, as it'd save me a bit of time.

Just also wanted to say I really appreciate all the work you've put into this repo, it's made life so much easier planning this project.

Thanks.

**Comment by jakka351** [comment, marked as answer]
Thanks for the feedback, appreciate it. The problem with completely removing the FDIM is that you lose functionality of the ICC buttons. I would recommend relocating the FDIM board somewhere out of sight so it can give the ICC buttons functionality. 

As for the code and emulation you want to do - I believe that is for replacing the ACM, not the FDIM. You lose radio and cd player functionality if you do that, so I think that was done with an aftermarket headunit going in.

I would recommend actually keeping all the modules and integrating the Pi with them via a can interface so you have all the factory functionality and the additional stuff with the pi.
